Source/santametricservice/Writers/SNTMetricHTTPWriter.m

```diff
@@ -18,103 +18,101 @@
 #import "Source/common/SNTLogging.h"
 #import "Source/santametricservice/Writers/SNTMetricHTTPWriter.h"
 
-@implementation SNTMetricHTTPWriter {
- @private
-  MOLAuthenticatingURLSession *_authSession;
-}
+@interface SNTMetricHTTPWriter ()
+@property SNTConfigurator *configurator;
+@end
+
+@implementation SNTMetricHTTPWriter
 
 - (instancetype)init {
   self = [super init];
   if (self) {
-    _authSession = [[MOLAuthenticatingURLSession alloc] init];
+    _configurator = [SNTConfigurator configurator];
   }
   return self;
 }
 
+- (MOLAuthenticatingURLSession *)createSessionWithHostname:(NSURL *)url
+                                                   Timeout:(NSTimeInterval)timeout {
+  NSURLSessionConfiguration *config = [NSURLSessionConfiguration ephemeralSessionConfiguration];
+  config.TLSMinimumSupportedProtocolVersion = tls_protocol_version_TLSv12;
+  config.HTTPShouldUsePipelining = YES;
+
+  config.timeoutIntervalForRequest = timeout;
+  config.timeoutIntervalForResource = timeout;
```

Comment:
super minor nit: it looks like `timeoutIntervalForResource` is all you need here - that should cover the full "connection initiated" to "data fully received" span. `..ForRequest` seems to only control the time allowed between _any_ bytes being received.